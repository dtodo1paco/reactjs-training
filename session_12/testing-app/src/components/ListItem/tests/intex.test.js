import React from 'react';
import { shallow } from 'enzyme';
import Component from '../index';

describe(`${Component.displayName} basics`, () => {
  it("should render correctly", () => {
    const wrapper = shallow(<Component />);
  });
  it("should match snapshot", () => {
    const wrapper = shallow(<Component />);
    expect(wrapper.getElements()).toMatchSnapshot();
  });
});
describe(`${Component.displayName} structure`, () => {
  it("should have a li", () => {
    const wrapper = shallow(<Component />);
    expect(wrapper.find('li').length).toEqual(1);
  });
  it("should render prop item as a li element", () => {
    const wrapper = shallow(<Component item="test"/>);
    expect(wrapper.find('li').text()).toEqual("test");
  });
});