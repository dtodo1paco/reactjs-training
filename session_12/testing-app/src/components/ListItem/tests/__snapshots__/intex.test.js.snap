// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem basics should match snapshot 1`] = `
Array [
  <li
    className="item"
  />,
]
`;
